packages/core/micro/src/sheetlet.ts

```diff
@@ -0,0 +1,871 @@
+
+    public setCellText(row: number, col: number, value: Primitive | undefined) {
+        // setting text clears any cell data
+        this.matrix.storeCellText(row, col, value);
+        this.invalidate(pointToKey(row, col));
+    }
+
+    public parseRef(text: string): Point | undefined {
+        return parseRef(this.matrix.numRows - 1, this.matrix.numCols - 1, text);
+    }
+
+    public getCell(row: number, col: number) {
+        return this.matrix.loadCellData(row, col) as Cell | undefined;
+    }
+
+    public readCell(row: number, col: number) {
+        let cell = this.getCell(row, col);
+        if (cell === undefined) {
+            cell = makeCell(this.matrix.loadCellText(row, col));
+            if (cell !== undefined) {
+                this.matrix.storeCellData(row, col, cell);
+            }
+        }
+        return cell;
+    }
+
+    public evaluateCell(row: number, col: number): Primitive | undefined {
+        const cell = this.readCell(row, col);
+        if (cell && cell.flag === CalcFlag.Dirty) {
+            try {
+                rebuild([pointToKey(row, col)], this);
+            } catch (e) {
+                console.error(`Rebuild failure: ${e}`);
+            }
+        }
+        if (cell) {
+            return isFormulaCell(cell) ?
+                cell.value === undefined ?
+                    undefined :
+                    this.primitiveFromValue([row, col], cell.value)
+                :
+                cell.content;
+        }
+        return undefined;
+    }
+
+    /**
+     * Evaluate a formula string that can start with or without
+     * '='. The resulting formula evaluates in the context of the
+     * whole sheet and creates no dependencies to the sheet (TODO). We
+     * assume that these formulas are run over clean sheets and will
+     * not calc dirty cells on demand. We return `undefined` if we
+     * encounter anything dirty during calc.
+     */
+    public evaluateFormula(formula: string): Primitive | undefined {
+        const program = formula[0] === "=" ? formula.substring(1) : formula;
+        const start: Point = [0, 0];
+        const end: Point = [this.matrix.numRows - 1, this.matrix.numCols - 1];
+        const fn = compile(program);
+        if (fn === undefined) { return undefined; }
+        const value = fn(undefined, new Range<any>(this.outOfSheetContext, start, end))[1];
+        return isDelayed(value) ? undefined : this.primitiveFromValue(undefined, value);
+    }
+
+    private primitiveFromValue<O>(origin: O, value: CalcValue<O>): Primitive | undefined {
+        switch (typeof value) {
+            case "number":
+            case "string":
+            case "boolean":
+                return value;
+            case "function":
+                return "<function>";
+            case "object":
+                if (value instanceof Range) {
+                    return this.primitiveFromValue(origin, value.read("value", origin) as CalcValue<O>);
+                }
```

Comment:
FYI - evaluating a formula like 'A1' would attempt to "stringify" Range.  I added this special case as a workaround.

Reply:
Good catch! I had ` coerceResult` for evaluating cells but forgot to use it here. Might be possible to unify the code by making ` coerceResul` generic.